api/src/main/java/org/openmrs/module/appointmentscheduling/api/AppointmentService.java

```diff
@@ -980,4 +981,40 @@ Integer getHistoryCountByConditions(Date fromDate, Date endDate,
     @Authorized(AppointmentUtils.PRIV_SCHEDULE_APPOINTMENTS)
 	Appointment bookAppointment(Appointment appointment, Boolean allowOverbook)
 			throws TimeSlotFullException;
+
+	/**
+	 * returns daily count for the statuses
+	 * scheduled
+	 * missed
+	 * completed
+	 */
+	@Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
+    List<AppointmentDailyCount> getAppointmentsDailyCount(Date fromDate,
+														  Date toDate, Location location, Provider provider, AppointmentType appointmentType,
+														  AppointmentStatus status, VisitType visitType);
+	/**
+	 * returns list of early appointment visits
+	 * who came late/early for their visits
+	 */
+	@Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
+	List<Appointment> getEarlyVisits(Date fromDate,
+								  Date toDate,  Location location, Provider provider, AppointmentType appointmentType,
+								  AppointmentStatus status, VisitType visitType) throws APIException;
+
+	/**
+	 * returns list of late appointments visits
+	 * who came late/early for their visits
+	 */
+	@Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
+	List<Appointment> getLateVisits(Date fromDate,
+											Date toDate,  Location location, Provider provider, AppointmentType appointmentType,
+											AppointmentStatus status, VisitType visitType) throws APIException;
+
+	/**
+	 * returns list of patient who have not come for the visits using
+	 * minDays and maxDays range provided
+	 */
+    @Authorized(AppointmentUtils.PRIV_VIEW_APPOINTMENTS)
+    List<Appointment> getDefaultersList(int minDays, int maxDays, Provider provider, AppointmentType type, VisitType visitType,
```

Comment:
I probably wouldn't use the term "Defaulters" here, as that has various meanings.  I'd call this what it is - getLateAppointments(...), where 2 of the options are minDaysLate and maxDaysLate.  I'd recommend that you consider changing from multiple methods that all support a wide range of method parameters to a single method parameter that you can extend over time to provide more search options:
getAppointments(AppointmentSearchCriteria);
where
AppointmentSearchCriteria is a simple bean that can be configured with:
* minDaysLate
* maxDaysLate
* minAppointmentDate
* maxAppointmentDate
* location
* provider
* appointmentStatus
* visitType
* ...etc

Reply:
would you happen to know a link I can read more about this? Kindly share